datalad/support/gitrepo.py

```diff
@@ -69,7 +72,10 @@ def _normalize_path(base_dir, path):
     # But we don't want to realpath relative paths, in case cwd isn't the correct base.
 
     if isabs(path):
-        path = realpath(path)
+        # path might already be a symlink pointing to annex etc,
+        # so realpath only its directory, to get "inline" with realpath(base_dir)
+        # above
+        path = opj(realpath(dirname(path)), basename(path))
```

Comment:
@bpoldrack this was a generic fix btw -- just didn't bother for a separate PR, if this one drags too long, we might want to pick it up separately